# Generated by Django 5.0.1 on 2024-02-09 23:21

from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ("heats", "0003_heat_pool"),
        ("locations", "0001_initial"),
        ("participants", "0011_remove_participant_unique_race_for_bib_number_and_more"),
        ("race", "0003_racetype_ftt_allowed_racetype_participants_allowed"),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.RemoveConstraint(
            model_name="participant",
            name="unique_race_for_user",
        ),
        migrations.RemoveConstraint(
            model_name="participant",
            name="unique_race_for_bib_number",
        ),
        migrations.AddConstraint(
            model_name="participant",
            constraint=models.UniqueConstraint(
                fields=("user", "race"),
                name="unique_race_for_user",
                violation_error_message="This user is already registered for this race!",
            ),
        ),
        migrations.AddConstraint(
            model_name="participant",
            constraint=models.UniqueConstraint(
                condition=models.Q(("is_active", True)),
                fields=("bib_number", "race"),
                name="unique_race_for_bib_number",
                violation_error_message="Another active participant is already using this bib number.",
            ),
        ),
    ]
